benches/howzit/src/howzit.rs

```diff
@@ -0,0 +1,212 @@
+        let random = rng.gen_range(0, max);
+
+        // subtract the random number from 2^PROBE_RANGE.end to invert the distribution
+        let inverse = 2u32.pow(*Self::PROBE_RANGE.end()) - random;
+
+        // take the log base 2 of the result to get the probe
+        // we'll use the leading zeros trick here
+        let probe = 32 - (inverse.leading_zeros() as u32);
+        
+        probe.into()
+
+    }
+
+}
+
+impl From<u32> for Probe {
+    fn from(value: u32) -> Self {
+        let value = (
+            value % Self::PROBE_RANGE.end()
+        ) + Self::PROBE_RANGE.start();
+        match value {
+            1 => Probe::Probe1,
+            2 => Probe::Probe2,
+            3 => Probe::Probe3,
+            _ => Probe::Probe1,
+        }
+    }
+}
+
+impl TryInto<Identifier> for Probe {
+    type Error = anyhow::Error;
+
+    fn try_into(self) -> Result<Identifier, Self::Error> {
+        match self {
+            Probe::Probe1 => Ok(Identifier::new("probe_1")?),
+            Probe::Probe2 => Ok(Identifier::new("probe_2")?),
+            Probe::Probe3 => Ok(Identifier::new("probe_3")?),
+        }
+    }
+}
+pub struct Howzit {
+    pub howzit_package_path : PathBuf,
+    pub wallet : Arc<RwLock<LocalAccount>>,
+    pub rest_client : Client,
+    faucet_client_url :Url,
+    pub faucet_client : FaucetClient,
+    pub faucet_auth_token : String
+}
+
+impl Clone for Howzit {
+    fn clone(&self) -> Self {
+        Howzit {
+            howzit_package_path: self.howzit_package_path.clone(),
+            wallet: self.wallet.clone(),
+            rest_client: self.rest_client.clone(),
+            faucet_client_url : self.faucet_client_url.clone(),
+            faucet_client: FaucetClient::new_from_rest_client(
+                self.faucet_client_url.clone(),
+                self.rest_client.clone()
+            ).with_auth_token(
+                self.faucet_auth_token.clone()
+            ),
+            faucet_auth_token : self.faucet_auth_token.clone()
+        }
+    }
+
+}
+
+impl Howzit {
+
+    /// Generates a new Howzit instance with a random wallet
+    pub fn generate(
+        howzit_package_path: PathBuf,
+        rest_client: Client,
+        faucet_client_url : Url,
+        faucet_auth_token : String
+    ) -> Self {
+        let wallet = LocalAccount::generate(&mut rand::rngs::OsRng);
+        Howzit { 
+            howzit_package_path, 
+            wallet: Arc::new(RwLock::new(wallet)),
+            rest_client : rest_client.clone(),
+            faucet_client_url : faucet_client_url.clone(),
+            faucet_client: FaucetClient::new_from_rest_client(
+                faucet_client_url,
+                rest_client
+            ).with_auth_token(
+                faucet_auth_token.clone()
+            ),
+            faucet_auth_token
+        }
+    }
+
+    /// Builds and publishes the howzit package
+    pub async fn build_and_publish(&self) -> Result<(), anyhow::Error> {
```

Comment:
Why not just make the `&mut self` receiver so you don't have to deal with locking?